### PR TITLE
remoteproc: implement rproc_virtio_read_config/rproc_virtio_write_config

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
so the rpmsg could access the configuration space as needed

Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
This patch split from https://github.com/OpenAMP/open-amp/pull/155 since it be commited first independently without the collaboration with kernel community

**Comment by wjliang** [comment]
Looks good to me, with small fix required.

File: lib/remoteproc/remoteproc_virtio.c [hunk 1]
```diff
@@ -126,20 +126,43 @@ static uint32_t rproc_virtio_negotiate_features(struct virtio_device *vdev,
...
-	(void)offset;
-	(void)src;
-	(void)length;
+	struct remoteproc_virtio *rpvdev;
+	struct fw_rsc_vdev *vdev_rsc;
+	struct metal_io_region *io;
+	char *config;
+
+	if (offset + length > vdev->config_len || offset + length < length)
+		return;
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply lib/remoteproc/remoteproc_virtio.c]: The check can catch the overflow error, we can remove the check if the caller come from trusted source.

**Comment by xiaoxiang781216** [comment]
@arnopo please review the updated patch which address @wjliang 's concern.

**Comment by xiaoxiang781216** [comment]
@arnopo cloud you guy spend some time to fix the build break?
image
The code quality seem to bad for me. Is there nobody take care the build break in Linaro org?

**Comment by arnopo** [comment]
> @arnopo could your guy spend some time to fix the build break?
> image
> The code quality seem to bad for me. Is any people inside Linaro org take care the build break?

iplease test with this PR sent yesterday https://github.com/OpenAMP/libmetal/pull/115

**Comment by arnopo** [comment]
> > @arnopo could your guy spend some time to fix the build break?
> > image
> > The code quality seem to bad for me. Is any people inside Linaro org take care the build break?
> 
> Please test with this PR sent yesterday [OpenAMP/libmetal#115](https://github.com/OpenAMP/libmetal/pull/115)

ok forget my comment, i integrated the fix in libmetal, now test is ok

**Comment by xiaoxiang781216** [comment]
@arnopo , thanks for fast response. The test is pass now, feel free to givre your feedback.

**Comment by arnopo** [comment]
> @arnopo , thanks for fast response. The test is pass now, feel free to givre your feedback.

We are now in release process for the v2020.04. I plan to review your  patches after that

**Comment by xiaoxiang781216** [comment]
> > @arnopo , thanks for fast response. The test is pass now, feel free to givre your feedback.
> 
> We are now in release process for the v2020.04. I plan to review your patches after that

Sure.

**Comment by xiaoxiang781216** [comment]
Move to https://github.com/OpenAMP/open-amp/pull/155

**Pull request closed without merging**